udata/core/spatial/api.py

```diff
@@ -71,11 +70,14 @@ def get(self):
 class ZonesAPI(API):
     @api.doc('spatial_zones',
              params={'ids': 'A zone identifiers list (comma separated)'})
+    @api.cache_page(check_serverside=False, key_pattern='zones%s',
+                    client_timeout=ONE_DAY, server_timeout=ONE_WEEK,
+                    make_response=api.make_response)
     @api.marshal_with(feature_collection_fields)
     def get(self, ids):
         '''Fetch a zone list as GeoJSON'''
         zones = GeoZone.objects.in_bulk(ids)
-        zones = [zones[id] for id in ids]
+        zones = [zones[id_] for id_ in ids if id_ in zones]
```

Comment:
As you wish, I'll not fight on this. Note that it fixes a bug we have in production ;)